torchx/lib/torchx/backend.ex

```diff
@@ -799,9 +810,6 @@ defmodule Torchx.Backend do
     {Nx.as_type(left, type), Nx.as_type(right, type)}
   end
 
-  defp maybe_broadcast_bin_args(_out_shape, %{shape: {}} = l, r), do: {from_nx(l), from_nx(r)}
-  defp maybe_broadcast_bin_args(_out_shape, l, %{shape: {}} = r), do: {from_nx(l), from_nx(r)}
```

Comment:
@polvalente after upgrading to libtroch 2.1.0, I saw a bunch of warnings like this:

```
[W Resize.cpp:35] Warning: An output with one or more elements was resized since it had shape [], which does not match the required output shape [3]. This behavior is deprecated, and in a future PyTorch release outputs will not be resized unless they have zero elements. You can explicitly reuse an out tensor t by resizing it, inplace, to zero elements with t.resize_(0). (function _resize_output_check)
```

An example where this happens is:

```elixir
Nx.logical_and(Nx.tensor(1), Nx.tensor([1, 2, 3]))
```

The warning only happens if the first operand is a scalar. Doing the actual broadcasting removes the warning.

Reply:
We do the broadcasting explicitly in StableHLO too, so that's fine I think.